cli/src/commands/create/dataset.rs

```diff
@@ -1,15 +1,30 @@
 use failchain::ResultExt;
+use lazy_static::lazy_static;
 use log::info;
+use regex::Regex;
 use reinfer_client::{
     resources::comment::EntityKind, Client, DatasetFullName, NewDataset, SourceIdentifier,
 };
 use structopt::StructOpt;
 
 use crate::errors::{ErrorKind, Result};
 
+lazy_static! {
+    static ref RX_FULL_NAME: Regex =
+        Regex::new("^[A-Za-z0-9-_.]{1,1024}/[A-Za-z0-9-_]{1,256}").unwrap();
+}
+
+/// Ensures the name provided conforms to the <owner>/<name> structure
+fn validate_dataset_name(s: String) -> std::result::Result<(), String> {
+    match RX_FULL_NAME.is_match(&s) {
+        true => Ok(()),
+        false => Err(s),
+    }
```

Comment:
I personally prefer `if x { foo } else { bar }` rather than matching on boolean.

Also possible is to use something along the lines of `failchain::ensure!(RX_FULL_NAME.is_match(&s), ...)` as you're returning an error on the bad case - but that's probably not very ergonomic until we migrate from `failchain` to `anyhow`.

Reply:
Yep, makes sense to just use an `if` here rather than a `match`.

I don't think that we can use `anyhow` here since the expected return type of a validator function for `structopt` is `Result<(), String>` and so a conversion to an `anyhow` Error doesn't work here.